src/main/java/seedu/address/logic/commands/localcourse/LocalCourseSearchCommand.java

```diff
@@ -0,0 +1,96 @@
+package seedu.address.logic.commands.localcourse;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Objects;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.SeplendidModel;
+import seedu.address.model.localcourse.LocalCodeContainsKeywordsPredicate;
+import seedu.address.model.localcourse.LocalCourseAttribute;
+import seedu.address.model.localcourse.LocalNameContainsKeywordsPredicate;
+import seedu.address.seplendidui.UiUtil;
+
+/**
+ * Finds and lists all localcourses in SEPlendid whose name contains any of the argument keywords.
+ * Keyword matching is case-insensitive.
+ */
+public class LocalCourseSearchCommand extends LocalCourseCommand {
+    public static final String ACTION_WORD = "search"; // Use "search" as the command word
+
+    public static final String MESSAGE_SUCCESS = "Local courses searched: %1$s";
+
+
+    public static final String LOCALCOURSE_SEARCH_MESSAGE_USAGE = COMMAND_WORD
+            + " : Search local courses by attributes - localcode and localname";
+
+    private final LocalCourseAttribute attribute;
+
+    private final LocalCodeContainsKeywordsPredicate codePredicate;
+    private final LocalNameContainsKeywordsPredicate namePredicate;
+
+    /**
+     * Creates a LocalCourseSearchCommand to sort the local course list based on the code predicate.
+     * @param codePredicate Predicate use for search
+     */
+    public LocalCourseSearchCommand(LocalCodeContainsKeywordsPredicate codePredicate) {
+        this.attribute = LocalCourseAttribute.LOCALCODE;
+        this.codePredicate = codePredicate;
+        this.namePredicate = null;
+    }
+
+    /**
+     * Creates a LocalCourseSearchCommand to sort the local course list based on the name predicate.
+     * @param namePredicate Predicate use for search
+     */
+    public LocalCourseSearchCommand(LocalNameContainsKeywordsPredicate namePredicate) {
+        this.attribute = LocalCourseAttribute.LOCALNAME;
+        this.codePredicate = null;
+        this.namePredicate = namePredicate;
+    }
+
+    @Override
+    public CommandResult execute(SeplendidModel model) throws CommandException {
+        requireNonNull(model);
+
+        model.searchLocalCourses(attribute, codePredicate, namePredicate);
+
```

Comment:
I think you need to call seplendidmodel.updateSearchList or getSearchList method here to update the list.